storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+  // the previously written values.
+  index_.reset(new SMAIndexSubBlock(*tuple_store_,
+                                    *index_description_,
+                                    false,
+                                    index_memory_.get(),
+                                    kIndexSubBlockSize));
+  EXPECT_FALSE(index_->requiresRebuild());
+}
+
+TEST_F(SMAIndexSubBlockTest, TestRebuild) {
+  // This test is checking to see if the min/max values are set correctly on
+  // a rebuild of the index.
+  // Index long, float type.
+  createIndex({0, 2}, kIndexSubBlockSize);
+  int min = 0, max = 9010, step = 10;
+  std::int64_t sum_0 = 0;
+  double sum_2 = 0;
+  for (std::size_t i = min; i <= static_cast<std::size_t>(max); i += step) {
+    generateAndInsertTuple(i, false, "suffix");
+    sum_0 += i;
+    sum_2 += (i * 0.25);
+  }
+  index_->rebuild();
+  SMAEntry* entry0 = getEntryForAttribute(0);
+  SMAEntry* entry2 = getEntryForAttribute(2);
+  // Entries should and must exist.
+  ASSERT_TRUE(entry0 != nullptr && entry2 != nullptr);
+
+  // Check entry validity.
+  EXPECT_TRUE(entry0->min_entry_ref.valid);
+  EXPECT_TRUE(entry2->min_entry_ref.valid);
+  EXPECT_TRUE(entry0->max_entry_ref.valid);
+  EXPECT_TRUE(entry2->max_entry_ref.valid);
+
+  // Check min ids and values.
+  EXPECT_EQ(0, entry0->min_entry_ref.entry_ref_tuple);
+  EXPECT_EQ(0, entry2->min_entry_ref.entry_ref_tuple);
+  EXPECT_TRUE(sma_test::longs_equal(entry0->min_entry_ref.value,
+                                    tuple_store_->getAttributeValueTyped(0, 0)));
+
+  EXPECT_TRUE(sma_test::floats_equal(entry2->min_entry_ref.value,
+                                     tuple_store_->getAttributeValueTyped(0, 2)));
+
+  // Check max ids and values.
+  tuple_id max_id = (max - min)/step;
+  EXPECT_EQ(max_id, entry0->max_entry_ref.entry_ref_tuple);
+  EXPECT_EQ(max_id, entry2->max_entry_ref.entry_ref_tuple);
+  EXPECT_TRUE(sma_test::longs_equal(entry0->max_entry_ref.value,
+                                    tuple_store_->getAttributeValueTyped(max_id, 0)));
+  EXPECT_TRUE(sma_test::floats_equal(entry2->max_entry_ref.value,
+                                     tuple_store_->getAttributeValueTyped(max_id, 2)));
+
+  // Check sums.
+  EXPECT_TRUE(sma_test::longs_equal(entry0->sum_aggregate, TypedValue(sum_0)));
+  EXPECT_TRUE(sma_test::doubles_equal(entry2->sum_aggregate, TypedValue(sum_2)));
+
+  // Check count.
+  EXPECT_EQ(static_cast<std::size_t>(max_id + 1), index_->getCount());
+}
+
+TEST_F(SMAIndexSubBlockTest, TestRebuildWithNulls) {
+  // Check that building an index with nullable attributes and null values
+  // behvaes correctly.
+
+  createIndex({0, 1, 2}, kIndexSubBlockSize);
+  int min = 0, max = 9010, step = 10;
+
+  // Attribute 1 will contain some nulls so keep its sum while inserting tuples.
+  std::int64_t sum_1 = 0;
+  for (std::size_t i = min; i <= static_cast<std::size_t>(max); i += step) {
+    bool insertNull = (i % 4) == 0;
```

Comment:
I usually declare such variables as ``const bool``, when I know for sure that it won't be changed later. It's just a good practice. You need not change it in this test case though.